snuba/query/snql/parser.py

```diff
@@ -1317,4 +1395,26 @@ def parse_snql_query(
 
     # Validating
     _post_process(query, VALIDATORS)
+
+    if isinstance(query, LogicalQuery):  # avoid composite for now
+        if is_in_experiment(query, "", "snql.sort.query.projects", None):
+            # Experiment with sorting the query to improve cache hit rate
+            sorting_rate = state.get_config("snql.sort.query.rate", 0.0)
+            sorting_rate = 0.0 if not isinstance(sorting_rate, float) else sorting_rate
+
+            timer = Timer("query.sorting")
+            if settings.TESTING or random.random() < sorting_rate:
+                assert isinstance(query, LogicalQuery)
```

Comment:
You have an `isinstance(query, LogicalQuery)` call at line 1399. Do you actually need this ?

Reply:
I'll remove this, I added it when I was trying to fix the typing issue I mentioned with `format_query`.